File: app/src/main/java/com/viafourasample/src/activities/splash/SplashActivity.java
package com.viafourasample.src.activities.splash;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.viafoura.sampleapp.R;
import com.viafourasample.src.activities.main.MainActivity;

public class SplashActivity extends AppCompatActivity {
    private SplashActivityViewModel viewModel = new SplashActivityViewModel();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        viewModel.setSplashActivityCallback(new SplashActivityViewModel.SplashActivityCallback() {
            @Override
            public void onLoadingCompleted() {
                goToMain();
            }
        });
    }

    private void goToMain(){
        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/viafourasample/src/activities/splash/SplashActivity.kt
package com.viafourasample.src.activities.splash

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.viafoura.sampleapp.R
import com.viafourasample.src.activities.main.MainActivity

class SplashActivity : AppCompatActivity() {
    private val viewModel = SplashActivityViewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        viewModel.splashActivityCallback =
            SplashActivityViewModel.SplashActivityCallback { goToMain() }
    }

    private fun goToMain() {
        startActivity(Intent(applicationContext, MainActivity::class.java))
    }
}
